training/model_management/src/azureml/model/mgmt/utils/common_utils.py

```diff
@@ -113,3 +116,34 @@ def create_namespace_from_dict(var: Any):
         for key, val in var.__dict__.items():
             var.__dict__[key] = create_namespace_from_dict(val)
     return var
+
+
+def init_tc():
+    """Initialize app insights telemetry client."""
+    global tc
+    if tc is None:
+        try:
+            tc = TelemetryClient("71b954a8-6b7d-43f5-986c-3d3a6605d803")
+        except Exception as e:
+            print(f"Exception while initializing app insights: {e}")
+
+
+def tc_log(message):
+    """Log message to app insights."""
+    global tc
+    try:
+        print(message)
+        tc.track_event(name="FM_import_pipeline_debug_logs", properties={"message": message})
+        tc.flush()
```

Comment:
Flush on every log is very expensive. Let us not do any flush so that the TelemetryClient automatically flushes when the buffer is full & when the process finishes.

Reply:
ok